pkg/dependencies/factory.go

```diff
@@ -403,6 +407,38 @@ func (f *Factory) WithProvider(clusterConfigFile string, clusterConfig *v1alpha1
 				f.dependencies.Kubectl,
 				time.Now,
 			)
+		case v1alpha1.NutanixDatacenterKind:
+			datacenterConfig, err := v1alpha1.GetNutanixDatacenterConfig(clusterConfigFile)
+			if err != nil {
+				return fmt.Errorf("unable to get datacenter config from file %s: %v", clusterConfigFile, err)
+			}
+
+			machineConfigs, err := v1alpha1.GetNutanixMachineConfigs(clusterConfigFile)
+			if err != nil {
+				return fmt.Errorf("unable to get machine config from file %s: %v", clusterConfigFile, err)
+			}
+
+			url := fmt.Sprintf("%s:%d", datacenterConfig.Spec.Endpoint, datacenterConfig.Spec.Port)
+			nutanixCreds := prismgoclient.Credentials{
+				URL:      url,
+				Username: os.Getenv("NUTANIX_USER"),
+				Password: os.Getenv("NUTANIX_PASSWORD"),
+				Endpoint: datacenterConfig.Spec.Endpoint,
+				Port:     fmt.Sprintf("%d", datacenterConfig.Spec.Port),
+			}
+			client, err := v3.NewV3Client(nutanixCreds)
+			if err != nil {
+				return fmt.Errorf("error creating nutanix client: %v", err)
+			}
```

Comment:
Can you move this section to a `With..()` func?
Maybe you can call it `WithPrismClient() *Factory` and make prism client a part of the `Dependencies` struct https://github.com/aws/eks-anywhere/blob/5dffa8f12fb2e666b4c04b9d5d450395d3ddad3b/pkg/dependencies/factory.go#L48

Also, can we check if the env vars are actually set using `os.LookupEnv` instead of `os.Getenv()`?